schemas/didDocument/didDocument.json

```
@@ -0,0 +1,40 @@
+{
+    "$id": "/did-core.didDocument",
+    "type": "object",
+    "properties": {
+      "@context": {
+        "title": "@context",
+        "description": "https://w3c.github.io/did-core-registry/#context",
+        "anyOf": [
+          { "type": "string" },
+          { "type": "array",
+            "items": [
+              {
+                "type": "string",
+                "const": "https://www.w3.org/ns/did/v1"
+              }
+            ] 
+          },
+          { "type": "object" }
+        ],
+        "required": false
+      },
+      "id": {
+        "title": "DID",
+        "description": "https://w3c.github.io/did-core-registry/#id",
+        "type": "string",
```

```suggestion
        "type": "string",
        "pattern": "^did:[a-z0-9]+:[A-Za-z0-9.-_:?=&;]+$",
```

This JSON Schema is too lax - if we're going to provide JSON Schema we should really lock it down and ensure it matches the ABNF in the spec... which probably means that we should have templates for URIs (at least).

---

> [A-Za-z0-9.-_:?=&;]

Also, I expect that what I suggested is still too lax... because you can do all sorts of invalid combinations still ... like did:foo:;=

---

@OR13 just a reminder to not close comments that you didn't raise... and in general, never close comments. It prevents other people from seeing the discussion, and /sometimes/ someone will pipe up and let us know that we made a mistake (or will provide a better fix).